packages/components/src/inputs/src/SearchField.module.css

```diff
@@ -0,0 +1,57 @@
+.hop-SearchField  {
+    --hop-SearchField-gap: var(--hop-space-stack-xs);
+    --hop-SearchField-default-inline-size: 20rem;
+    --hop-SearchField-color: var(--hop-neutral-text-weak);
+    --hop-SearchField-padding: var(--hop-space-inset-md);
+
+    display: flex;
+    flex-direction: column;
+    gap: var(--hop-SearchField-gap);
+    inline-size: var(--hop-SearchField-default-inline-size);
+}
+
+.hop-SearchField--fluid  {
+    inline-size: 100%;
+}
+
+.hop-SearchField__Label {
+    order: 0;
+
+    font-family: var(--hop-heading-xs-font-family);
+    font-size: var(--hop-heading-xs-font-size);
+    font-weight: var(--hop-heading-xs-font-weight);
+    line-height: var(--hop-heading-xs-line-height);
+    color: var(--hop-SearchField-color);
+}
+
+.hop-SearchField__prefix {
+    color: var(--hop-SearchField-color);
+}
+
+.hop-SearchField__InputGroup {
+    order:1;
+    inline-size: 100%;
+    padding-inline: var(--hop-SearchField-padding);
+}
+
+.hop-SearchField--clearable .hop-SearchField__InputGroup {
+    padding-inline-end: 0;
+}
+
+.hop-SearchField__ErrorMessage,
+.hop-SearchField__HelperMessage {
+    order: 3;
+}
+
+.hop-SearchField input::-webkit-search-cancel-button,
+.hop-SearchField input::-webkit-search-decoration {
+    appearance: none;
+}
+
+.hop-SearchField__ClearButton {
+    flex-shrink: 0;
+}
+
+.hop-SearchField[data-empty] .hop-SearchField__ClearButton {
+    display: none;
```

Comment:
hmm you could have internal variables for this too.
so --display: none; is set inside .hop-SearchField[data-empty]
and display: var(--display) is set inside .hop-SearchField__ClearButton

Reply:
since i'm toggling a default value, i feel like this isn'T necessary

Reply:
the internal variable would allow you to cut down on selectors. The value is updated and with an internal value, it would be clear that the display is changing.